File: backend/src/main/java/com/boatsched/entity/Schedule.java
package com.boatsched.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;
import java.time.LocalTime;

@Entity
@Table(name = "schedules")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Schedule {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @Column(nullable = false)
    private LocalDate scheduleDate;

    @Column(nullable = false)
    private LocalTime scheduleTime;

    @Column(nullable = false)
    private Integer paxCount;

    @Column(nullable = false)
    private String purpose;

    private String note;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private Status status = Status.REQUESTED;

    public enum Status {
        REQUESTED, APPROVED, CANCELED
    }
}

File: frontend/lib/models/schedule.dart
class Schedule {
  final int id;
  final int userId;
  final String storeName;
  final String representativeName;
  final String phoneNumber;
  final DateTime scheduleDate;
  final String scheduleTime;
  final int paxCount;
  final String purpose;
  final String? note;
  final String status;

  Schedule({
    required this.id,
    required this.userId,
    required this.storeName,
    required this.representativeName,
    required this.phoneNumber,
    required this.scheduleDate,
    required this.scheduleTime,
    required this.paxCount,
    required this.purpose,
    this.note,
    required this.status,
  });

  factory Schedule.fromJson(Map<String, dynamic> json) {
    return Schedule(
      id: json['id'],
      userId: json['userId'],
      storeName: json['storeName'],
      representativeName: json['representativeName'],
      phoneNumber: json['phoneNumber'],
      scheduleDate: DateTime.parse(json['scheduleDate']),
      scheduleTime: json['scheduleTime'],
      paxCount: json['paxCount'],
      purpose: json['purpose'],
      note: json['note'],
      status: json['status'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'scheduleDate': scheduleDate.toIso8601String().split('T')[0],
      'scheduleTime': scheduleTime,
      'paxCount': paxCount,
      'purpose': purpose,
      'note': note,
    };
  }

  bool get isRequested => status == 'REQUESTED';
  bool get isApproved => status == 'APPROVED';
  bool get isCanceled => status == 'CANCELED';
}
